libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java

```diff
@@ -607,13 +611,16 @@ private void handleResendRequest(FIXMessage message) throws IOException {
                 return;
             }
 
-            FIXValue endSeqNo = message.valueOf(EndSeqNo);
-            if (endSeqNo == null) {
+            FIXValue value = message.valueOf(EndSeqNo);
+            if (value == null) {
                 sendReject(message.getMsgSeqNum(), RequiredTagMissing, "EndSeqNo(16) not found");
                 return;
             }
 
-            sendSequenceReset(beginSeqNo, endSeqNo.asInt() + 1);
+            long endSeqNo = value.asInt();
+            long newSeqNo = endSeqNo == 0 ? txMsgSeqNum + 1 : endSeqNo + 1;
```

Comment:
set newSeqNo to txMsgSeqNum. Also need to verify that if they are requesting sequences > txMsgSeqNum then limit to maxSeqNum

Reply:
If the BeginSeqNo(7) value exceeds `txMsgSeqNum`, we should probably respond with a Reject(3) message, right? 🤔 

If the EndSeqNo(16) value exceeds `txMsgSeqNum`, we could probably respond with a SequenceReset(4) Gap Fill message with a NewSeqNo(36) value equal to `txMsgSeqNum`, as you suggest. 👍🏼

Reply:
yes if BeginSeqNo > txMsgSeqNum then should reject

Reply:
These two cases are now implemented.